src/sickle.ts

```diff
@@ -36,12 +43,15 @@ class Annotator {
 
   constructor(private options: SickleOptions) { this.indent = 0; }
 
-  annotate(file: ts.SourceFile): string {
+  annotate(file: ts.SourceFile): SickleOutput {
     this.output = [];
     this.file = file;
     this.visit(file);
     this.assert(this.indent == 0, 'visit() failed to track nesting');
-    return this.output.join('');
+    return {
+      output: this.output.join(''),
+      externs: null,
+    };
```

Comment:
sounds good, just making sure we are on the same page. We can learn more about the downsides of this pattern as the codebase grows.